lib/routes/ieee/journal.ts

```diff
@@ -2,89 +2,77 @@ import { Route } from '@/types';
-    const isnumber = response.currentIssue.issueNumber;
-    const jrnlName = response.displayTitle;
+    const metadata = await fetchMetadata(publicationNumber);
+    const { displayTitle, currentIssue, preprintIssue, coverImagePath } = metadata;
+    const { issueNumber, volume } = earlyAccess ? preprintIssue : currentIssue;
 
-    const response2 = await got
-        .post(`${host}/rest/search/pub/${punumber}/issue/${isnumber}/toc`, {
-            cookieJar,
-            json: {
-                punumber,
-                isnumber,
-                sortType,
-                rowsPerPage: '100',
-            },
-        })
-        .json();
-    let list = response2.records.map((item) => {
-        const $2 = load(item.articleTitle);
-        const title = $2.text();
-        const link = item.htmlLink;
-        const doi = item.doi;
-        let authors = 'Do not have author';
-        if (Object.hasOwn(item, 'authors')) {
-            authors = item.authors.map((itemAuth) => itemAuth.preferredName).join('; ');
-        }
-        let abstract = '';
-        Object.hasOwn(item, 'abstract') ? (abstract = item.abstract) : (abstract = '');
-        return {
-            title,
-            link,
-            authors,
-            doi,
-            volume,
-            abstract,
-        };
-    });
+    const tocData = await fetchTOCData(publicationNumber, issueNumber);
+    const list = tocData.records.map((item) => {
+        const mappedItem = mapRecordToItem(volume)(item);
 
-    const renderDesc = (item) =>
-        art(path.join(__dirname, 'templates/description.art'), {
-            item,
+        mappedItem.description = art(path.join(__dirname, 'templates/description.art'), {
+            item: mappedItem,
         });
-    list = await Promise.all(
-        list.map((item) =>
-            cache.tryGet(item.link, async () => {
-                if (item.abstract !== '') {
-                    const response3 = await got(`${host}${item.link}`);
-                    const { abstract } = JSON.parse(response3.body.match(/metadata=(.*);/)[1]);
-                    const $3 = load(abstract);
-                    item.abstract = $3.text();
-                    item.description = renderDesc(item);
-                }
-                return item;
-            })
-        )
-    );
+
+        return mappedItem;
+    });
 
     return {
-        title: jrnlName,
-        link: jrnlUrl,
+        title: displayTitle,
+        link: `${ieeeHost}/xpl/tocresult.jsp?isnumber=${issueNumber}`,
         item: list,
+        image: `${ieeeHost}${coverImagePath}`,
     };
 }
+
+async function fetchMetadata(punumber) {
+    const response = await got(`${ieeeHost}/rest/publication/home/metadata?pubid=${punumber}`, {
+        cookieJar,
+    });
+    return response.data;
+}
+
+async function fetchTOCData(punumber, isnumber) {
+    const response = await got.post(`${ieeeHost}/rest/search/pub/${punumber}/issue/${isnumber}/toc`, {
+        cookieJar,
```

Comment:
`cookieJar` is now useless since `@/utils/got` will not place cookies from `set-cookie` into cookieJar.